### ISSUE TITLE
pipe the lcov output

### ISSUE THREAD

**Issue by mathiasrw** [nickmerwin/node-coveralls#114]
> Once your app is instrumented for coverage, and building, you need to pipe the lcov output to ./node_modules/coveralls/bin/coveralls.js.

Would be awesome if you could share an example.

**Comment by pocesar** [comment]
usually this suffice `cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js`

**Comment by ETHproductions** [comment]
@pocesar When I try to run the example Istanbul command, everything works fine until it tries to perform `cat`. I'm on a Windows computer, so there is no `cat` command. I read somewhere that you can replace it with `type`, but that just results in this error message:

    '.' is not recognized as an internal or external command,
    operable program or batch file.

Any idea what I should do?

**Comment by pocesar** [comment]
you can install git-scm, that comes with a lot of 'linux shell' utilities that can be used in windows. other than that you could create a "proxy" .js file that takes file input then streams to output like `require('fs').createReadStream(process.argv[1]).pipe(process.stdout)` then call it `node proxy.js  ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js` (or skip the command line pipe altogether and spawn coveralls directly from the js file)

**Comment by ETHproductions** [comment]
First I tried just putting the paths into proxy.js like so:

    require('fs')
        .createReadStream('./coverage/lcov.info')
        .pipe('./node_modules/coveralls/bin/coveralls.js');

That raises an error though, so I tried creating a WriteStream out of the coveralls file:

    var fs = require('fs');
    fs.createReadStream('./coverage/lcov.info')
      .pipe(fs.createWriteStream('./node_modules/coveralls/bin/coveralls.js'));

However, that just copies the *contents* of proxy.js to coveralls.js. I also tried the exact code you mentioned, but that raises the error `'.' is not recognized as an external or internal command`. Any thoughts?

Also, is the `istanbul cover ...` command supposed to be placed in the package.json scripts as `test`, or as `coverage`, or in `after-success:` in the .travis.yml file?

**Comment by ETHproductions** [comment]
Would just calling `node ./node_modules/coveralls/bin/coveralls.js ./coverage/lcov.info` work instead?

**Comment by addaleax** [comment]
@ETHproductions no, but `node ./node_modules/coveralls/bin/coveralls.js < ./coverage/lcov.info` would be virtually equivalent to `cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js`.

**Comment by ETHproductions** [comment]
I decided to give up on trying to get it to work on *my* computer, since Travis CI only tests on Windows and Linux. After putting the example Istanbul command after `after_success:`, it finally worked on Travis :smile:

@addaleax I tried your suggestion, but coverage.js throws a 422 error ("Couldn't find a repository matching this job.")

**Comment by ankit213** [comment]
used this command in windows "./node_modules/coveralls/bin/coveralls.js < ./coverage/lcov.info" throw error  "Couldn't find a repository matching this job."
